Source/Shaders/BillboardCollectionFS.glsl

```diff
@@ -48,11 +57,73 @@ float getGlobeDepth(vec2 adjustedST, vec2 depthLookupST, bool applyTranslate, ve
 }
 #endif
 
+
+#ifdef SDF
+
+// Get the distance from the edge of a glyph at a given position sampling an SDF texture.
+float getDistance(vec2 position)
+{
+    return texture2D(u_atlas, position).r;
+}
+
+// Samples the sdf texture at the given position and produces a color based on the fill color and the outline.
+vec4 getSDFColor(vec2 position, float outlineWidth, vec4 outlineColor, float smoothing)
+{
+    float distance = getDistance(position);
+
+    if (outlineWidth > 0.0)
+    {
+        // Don't get the outline edge exceed the SDF_EDGE
+        float outlineEdge = clamp(SDF_EDGE - outlineWidth, 0.0, SDF_EDGE);
+        float outlineFactor = smoothstep(SDF_EDGE - smoothing, SDF_EDGE + smoothing, distance);
+        vec4 sdfColor = mix(outlineColor, v_color, outlineFactor);
+        float alpha = smoothstep(outlineEdge - smoothing, outlineEdge + smoothing, distance);
+        return vec4(sdfColor.rgb, sdfColor.a * alpha);
+    }
+    else
+    {
+        float alpha = smoothstep(SDF_EDGE - smoothing, SDF_EDGE + smoothing, distance);
+        return vec4(v_color.rgb, v_color.a * alpha);
+    }
+}
+#endif
+
 void main()
 {
     vec4 color = texture2D(u_atlas, v_textureCoordinates);
+
+#ifdef SDF
+
+    float outlineWidth = v_outlineWidth;
+    vec4 outlineColor = v_outlineColor;
+
+    // Get the current distance
+    float distance = getDistance(v_textureCoordinates);
+    float smoothing = fwidth(distance);
```

Comment:
I think this will need to be:

```glsl
#ifdef GL_OES_standard_derivatives
    float smoothing = fwidth(distance);
#else
    // Some alternative if GL_OES_standard_derivatives is not supported
#endif
```

Reply:
Same with `sampleOffset` below.

Reply:
Pushed a fallback that just does a single sample with a fixed smoothing.